Ical.Net/Evaluation/Evaluator.cs

```diff
@@ -17,33 +17,41 @@ protected void IncrementDate(ref CalDateTime dt, RecurrencePattern pattern, int
         if (interval == 0)
             return;
 
-        var old = dt;
-        switch (pattern.Frequency)
+        try
         {
-            case FrequencyType.Secondly:
-                dt = old.AddSeconds(interval);
-                break;
-            case FrequencyType.Minutely:
-                dt = old.AddMinutes(interval);
-                break;
-            case FrequencyType.Hourly:
-                dt = old.AddHours(interval);
-                break;
-            case FrequencyType.Daily:
-                dt = old.AddDays(interval);
-                break;
-            case FrequencyType.Weekly:
-                dt = DateUtil.AddWeeks(old, interval, pattern.FirstDayOfWeek);
-                break;
-            case FrequencyType.Monthly:
-                dt = old.AddDays(-old.Day + 1).AddMonths(interval);
-                break;
-            case FrequencyType.Yearly:
-                dt = old.AddDays(-old.DayOfYear + 1).AddYears(interval);
-                break;
-            // FIXME: use a more specific exception.
-            default:
-                throw new Exception("FrequencyType.NONE cannot be evaluated. Please specify a FrequencyType before evaluating the recurrence.");
+            var old = dt;
+            switch (pattern.Frequency)
+            {
+                case FrequencyType.Secondly:
+                    dt = old.AddSeconds(interval);
+                    break;
+                case FrequencyType.Minutely:
+                    dt = old.AddMinutes(interval);
+                    break;
+                case FrequencyType.Hourly:
+                    dt = old.AddHours(interval);
+                    break;
+                case FrequencyType.Daily:
+                    dt = old.AddDays(interval);
+                    break;
+                case FrequencyType.Weekly:
+                    dt = DateUtil.AddWeeks(old, interval, pattern.FirstDayOfWeek);
+                    break;
+                case FrequencyType.Monthly:
+                    dt = old.AddDays(-old.Day + 1).AddMonths(interval);
+                    break;
+                case FrequencyType.Yearly:
+                    dt = old.AddDays(-old.DayOfYear + 1).AddYears(interval);
+                    break;
+                // FIXME: use a more specific exception.
+                default:
+                    throw new Exception("FrequencyType.NONE cannot be evaluated. Please specify a FrequencyType before evaluating the recurrence.");
+            }
+        }
+        catch (ArgumentOutOfRangeException)
```

Comment:
Sticking to the ical.net definitions, isn't is about the Occurrence of the Event here?
How about sth. like
```cs
catch (ArgumentOutOfRangeException)
{
    // intentionally don't include the outer exception
    throw new EvaluationOutOfRangeException(
        $"""
         Evaluation aborted: An event occurrence starting at {dt} plus the next interval exceeds the maximum supported date/time value. 
         This may be caused by a recurrence pattern with a missing or excessive COUNT, or an UNTIL date missing or beyond the supported range.
         """
    );
}
```

Reply:
> Sticking to the ical.net definitions, isn't is about the Occurrence of the Event here?

Are you referring to 'recurrence' vs 'occurrence' in the the message in the `default` case? Actually I didn't touch that one, but to my understanding recurrences are occurrences that are produced by recurring elements, i.e. by RRULEs, right? Acutally this method is only used with RRULEs, so this part of the message could be ok. But the whole method is not well suited in this class. It should either be moved to a utility class or to `RecurrencePatternEvaluator`. Happen to have related work prepared in a different, WIP PR.

Will try to improve the message in the `EvaluationOutOfRangeException`

Reply:
> Evaluation aborted: An event occurrence starting at {dt} plus the next interval exceeds the maximum supported date/time value. This may be caused by a recurrence pattern with a missing or excessive COUNT, or an UNTIL date missing or beyond the supported range.

Hm, this is also not fully precise. Actually there isn't necessarily an occurrence at `dt`. I feel that the actual value also doesn't add too much value. Usually the problem here is just that people try to iterate an unbounded RRULE to its end, which isn't possible. The actual value, where the overflow happens, seems to be less relevant. Will try to find something.

Reply:
> Actually there isn't necessarily an occurrence

agree. 

The only intention regarding the messages is to make people see themselves what caused the exception and what they can do about it, instead of opening a ticket. Any wording serving this goal is fine.